Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2594,39 +2669,51 @@ - (void)tappedUnknownContactBlockOfferMessage:(OWSUnknownContactBlockOfferMessag
                                  style:UIAlertActionStyleDestructive
                                handler:^(UIAlertAction *_Nonnull action) {
                                    DDLogInfo(@"%@ Blocking an unknown user.", self.tag);
-                                   [self.blockingManager addBlockedPhoneNumber:errorMessage.contactId];
-                                   // Delete the block offer.
+                                   [self.blockingManager addBlockedPhoneNumber:interaction.contactId];
+                                   // Delete the offers.
                                    [self.editingDatabaseConnection
                                        readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-                                           [errorMessage removeWithTransaction:transaction];
+                                           contactThread.hasDismissedOffers = YES;
```

Comment:
Note that we now explicitly flag `TSContactThread` to avoid showing the offers again, if they've already been dismissed.